credentials/credentials-route.js

```diff
@@ -0,0 +1,233 @@
+const AWS = require('aws-sdk');
+var encryptor = require('file-encryptor');
+var fs= require('fs');
+const multer = require('multer');
+require("dotenv").config();
+const path = require('path');
+
+const s3 = new AWS.S3({
+  accessKeyId: process.env.REACT_APP_AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.REACT_APP_AWS_SECRET_ACCESS_KEY,
+  region: process.env.REACT_APP_AWS_REGION
+});
+console.log("access id", process.env.REACT_APP_AWS_ACCESS_KEY_ID)
+module.exports = (app) => {
+
+const uploadDirectory = './uploads';
+if (!fs.existsSync(uploadDirectory)) {
+  fs.mkdirSync(uploadDirectory);
+}
+  // Configure Multer storage
+const storage = multer.diskStorage({
+  destination: (req, file, cb) => {
+    cb(null, 'uploads/');
+  },
+  filename: (req, file, cb) => {
+    const fileName = file.originalname.toLowerCase().split(' ').join('-');
+    cb(null, fileName);
+  }
+});
+
+// Create a Multer instance
+const upload = multer({ storage });
+
+app.post('/upload', upload.single('file'), (req, res) => {
+  
+    try {
+      const file = req.file;
+      const fileName = req.body.fileName;
+      const extension = path.extname(fileName);
+      const filenameWithoutExtension = path.basename(fileName, extension);
+      const key = req.body.key;
+      
+
+      if (!file) {
+        return res.status(400).json({ error: 'No file provided' });
+      }
+
+      encryptor.encryptFile(req.file.path, filenameWithoutExtension+'-'+key+'.dat', key, function(err) {
+        const encryptedFile = fs.readFileSync('./'+filenameWithoutExtension+'-'+key+'.dat');
+        const params = {
+          Bucket: process.env.REACT_APP_BUCKET_NAME,
+          Key: filenameWithoutExtension+key,
```

Comment:
how do we make sure this key is unique?

Reply:
Made the key unique by generating a unique key using nanoid lib